File: Easy/Reach a given score/reach-a-given-score.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

public class GFG {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t=in.nextInt();
        while(t > 0)
        {
            int n = in.nextInt();
            Geeks obj = new Geeks();
            System.out.println(obj.count(n));
            t--;
        }
    }
}
// } Driver Code Ends


// Complete this function!

class Geeks {
    public long count(int n) {
        int dp[] = new int[n+1];
        int coins[] = {3, 5, 10};
        dp[0] = 1;
        
        for(int i = 0; i < coins.length; i++) {
            for(int j = coins[i]; j <= n; j++)
                dp[j] += dp[j - coins[i]];
            
        }
        return dp[dp.length-1];
    }
}

File: Easy/Reach a given score/reach-a-given-score.py
#User function Template for python3

class Solution:
    def count(self, n: int) -> int:
        dp = [0] * (n+1)
        dp[0] = 1
        for coins in [3, 5, 10]:
            for i in range (coins, n+1):
                dp[i] += dp[i-coins]
        return dp[-1]
        
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
    for _ in range(int(input())):
        n = int(input())
        ob = Solution()
        print(ob.count(n))
        
# } Driver Code Ends